### PR TITLE
Feature/cors

### PR DESCRIPTION
**Pull request by vbence86**
### Intent
To add functionality that tests the XHR instance when a network error occurs and gives an educated guess whether the XHR request was terminated by CORS issue.

### Solution
Provided ```responseURL``` remains empty, but ```readyState``` is ```4``` we assume is is blocked request due to cross browser origin issue.

### Tests
-  New unit test to cover this scenario automatically
- Tested manually against Safari, Chrome, FF, Edge

#### Notes
While reproducing the mentioned scenario in IE11, the request wasn't blocked and the 302 was appropriately processed. Nevertheless the content wasn't flipped as expected.

### CODE REVIEW

File: test/Ajax.js [hunk 1]
```diff
@@ -195,6 +198,16 @@ describe('Ajax', function() {
 			this.requests[0].error();
 		});
 
+		it('should fail when attempting to fetch resource from cross domain', function(done) {			
+			this.xhr.restore();
```
Comment (jbalsas) [inline test/Ajax.js]: Why is this needed? Could you add a comment here so we know?

Comment (vbence86) [inline test/Ajax.js]: FakeXMLHttpRequest cannot support CORS activities and therefore cannot trigger or imitate CORS exceptions. Here the test restores the original XMLHttpRequest object that will be used throughout this test scenario.

File: src/Ajax.js [hunk 2]
```diff
@@ -63,7 +63,11 @@ class Ajax {
 				resolve(request);
 			};
 			request.onerror = function() {
-				var error = new Error('Request error');
+				var message = 'Request error';
+				if (4 === request.readyState && !request.responseURL) {
```
Comment (eduardolundgren) [inline src/Ajax.js]: IE doesn't set responseURL after a 302 redirect so this case can be understood as CORS error. Make sure to test it.

**Comment by fernandosouza** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Comment by jbalsas** [comment]
We have verified this approach works as well in IE, but @vbence86 has come up with a more robust way to catch the CORS redirect issue. 

Based on browser observed behaviour, on CORS redirect errors, `xhr.readyState` jumps from `1` to `4` directly, which could probably be used to identify this case. @vbence86 is going to do some more thorough testing and send a PR with those changes if satisfactory.
